### DISCUSSION TITLE
Most effective way to shatter a board?

### DISCUSSION THREAD

**Discussion by patrikpatrik** [DanielChappuis/reactphysics3d#291] [General]
What would be the most effective way of shattering or destroying a board into multiple pieces? I've been thinking about this for a while now but wasn't quite sure how to implement it?

Let's say I have a ball that's thrown at a wall, and I want it to randomly shatter at impact and pieces of it falling. I did something like this years ago on Unity, but it had many steps one of which was to create a predefined shattered model of the same model on blender, and then import that and have it ready, then when the ball hit the original model replace it with the shattered model version but give collision meshes on each of those shattered pieces. While it did work I always thought there was another way without these predefined steps?

Would there be an alternate method for ReactPhysics3d?

**Comment by patrikpatrik** [comment]
I suppose no one really knows or wants to answer, and as usual I'm answering my own question. However, after days of intense research, I came to the conclusion that one of many methods is using tessellation shaders, raycasting, a geometry shader, and lastly a physics library such as bulletPhysics, ReactPhysics3D etc.. to perform what is known as 'dynamic destruction'.

Well, hopefully this won't be as painful as figuring out Skeletal Animations and linear algebra for several months...but down the rabbit hole I go~ Wish me luck!

**Comment by DanielChappuis** [comment]
Hello.

Some physics engines support Convex Decomposition. What you can try is to separate your board in several convex shapes (several rigid bodies) that are linked together with fixed joints. You can then break the joints if their reaction force/torque is large enough to be broken to separate the pieces. You can also separate the joints manually when some of the parts are hit.

**Reply by patrikpatrik** [reply]
That's a great idea actually! Does `ReactPhysics3D` support Convex decomposition? The only downside is that it's pre broken right? In other words the places where it's supposed to separate isn't random and always fixed? Either way it would be interesting to implement as an alternate option to real time destruction.
